src/dbnode/storage/series/reader.go

```diff
@@ -126,39 +138,45 @@ func (r Reader) readersWithBlocksMapAndBuffer(
 					return nil, err
 				}
 				if streamedBlock.IsNotEmpty() {
-					results = append(results, []xio.BlockReader{streamedBlock})
+					resultsBlock = append(resultsBlock, streamedBlock)
 					// NB(r): Mark this block as read now
 					block.SetLastReadTime(now)
 					if r.onRead != nil {
 						r.onRead.OnReadBlock(block)
 					}
 				}
-				continue
+				retrievedFromCache = true
```

Comment:
Hm interesting, I see the issue.  Quite "oh dear".

Reply:
Haha yeah this was a fun one to debug...